pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -2024,6 +2027,75 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         }
     }
 
+    template <class PrecisionT, class ParamT>
+    static void
+    applyPCPhase([[maybe_unused]] std::complex<PrecisionT> *arr,
+                 [[maybe_unused]] std::size_t num_qubits,
+                 [[maybe_unused]] const std::vector<std::size_t> &wires,
+                 [[maybe_unused]] bool inverse, [[maybe_unused]] ParamT angle,
+                 [[maybe_unused]] ParamT dim) {
+
+        const PrecisionT phase = inverse ? -angle : angle;
+        const std::complex<PrecisionT> upper_complex = {std::cos(phase),
+                                                        std::sin(phase)};
+        const std::complex<PrecisionT> lower_complex = {std::cos(phase),
+                                                        -std::sin(phase)};
+
+        // Matrix
+        const std::size_t matrix_dim = Pennylane::Util::exp2(wires.size());
+
+        std::vector<std::complex<PrecisionT>> matrixPCPhase(matrix_dim *
```

Comment:
Wondering if there is any concern on the scalability/performance of this implementation.